src/common/matmul.cpp

```
@@ -97,6 +97,13 @@ status_t matmul_attr_check(const matmul_desc_t &desc, const engine_t *engine,
     int dst_qmask_M = src_qmask_K;
     int dst_qmask_N = wei_qmask_N;
 
+    int per_tensor_mask = 0xfff;
+    int per_tensor_eq = 0x0;
+
+    for (int i = 0; i < ndims_src; ++i) {
+        per_tensor_eq |= (1 << i);
```

`per_tensor_eq` looks same as `per_tensor_mask` but variadic based on `ndims` value. Seems to me the latter is not needed. Could you confirm?

---

For ndims=3 they are equal. For ndims=2 `0xfff` makes no sense. For ndims=4 I'm not sure mask 7 works properly (though I might be wrong).

---

The reason `0xfff` is included is that is what benchdnn produces when you run with `per_tensor` policy for attributes, regardless of dimension. I want to cover any mask that is either the default `per_tensor` mask from benchdnn (`0xfff`) or an equivalent mask for any number of dimensions, so not only 7.